crates/matrix-sdk/src/test_utils/mocks.rs

```diff
@@ -980,6 +1157,20 @@ impl<'a> MockEndpoint<'a, RoomEventEndpoint> {
     }
 }
 
+/// A prebuilt mock for the `/messages` endpoint.
+pub struct RoomMessagesEndpoint;
+
+/// A prebuilt mock for getting a room messages in a room.
+impl<'a> MockEndpoint<'a, RoomMessagesEndpoint> {
+    /// Returns a messages endpoint that emulates success, i.e. the messages
+    /// provided as `response` could be retrieved.
+    pub fn ok(self, response: impl Into<Value>) -> MatrixMock<'a> {
```

Comment:
No, we should not use `impl Into<Value>`, otherwise we lose all the usefulness of these mocking APIs by reverting to "we don't know what the return value looks like, it can be an arbitrary JSON". Instead we should use high-level types. We have another place where we do mock messages, grep for `fn mock_messages` (…it's even better, we do have two implementations :D).